contracts/src/v0.1/RequestResponseCoordinator.sol

```diff
@@ -521,4 +486,199 @@ contract RequestResponseCoordinator is
+    }
+
+    function fulfillDataRequestBool(
+        uint256 requestId,
+        bool response,
+        RequestCommitment memory rc,
+        bool isDirectPayment
+    ) external nonReentrant returns (uint256) {
+        uint256 startGas = gasleft();
+        validateDataResponse(rc, requestId);
+        RequestResponseConsumerBase rr;
+        bytes memory resp = abi.encodeWithSelector(
+            rr.rawFulfillDataRequestBool.selector,
+            requestId,
+            response
+        );
+        bool success = fulfill(resp, rc);
+        uint256 payment = pay(rc, isDirectPayment, startGas);
+        emit DataRequestFulfilledBool(requestId, response, payment, success);
+        return payment;
+    }
+
+    function fulfillDataRequestString(
+        uint256 requestId,
+        string memory response,
+        RequestCommitment memory rc,
+        bool isDirectPayment
+    ) external nonReentrant returns (uint256) {
+        uint256 startGas = gasleft();
+        validateDataResponse(rc, requestId);
+        RequestResponseConsumerBase rr;
+        bytes memory resp = abi.encodeWithSelector(
+            rr.rawFulfillDataRequestString.selector,
+            requestId,
+            response
+        );
+        bool success = fulfill(resp, rc);
+        uint256 payment = pay(rc, isDirectPayment, startGas);
+        emit DataRequestFulfilledString(requestId, response, payment, success);
+        return payment;
+    }
+
+    function fulfillDataRequestBytes32(
+        uint256 requestId,
+        bytes32 response,
+        RequestCommitment memory rc,
+        bool isDirectPayment
+    ) external nonReentrant returns (uint256) {
+        uint256 startGas = gasleft();
+        validateDataResponse(rc, requestId);
+        RequestResponseConsumerBase rr;
+        bytes memory resp = abi.encodeWithSelector(
+            rr.rawFulfillDataRequestBytes32.selector,
+            requestId,
+            response
+        );
+        bool success = fulfill(resp, rc);
+        uint256 payment = pay(rc, isDirectPayment, startGas);
+        emit DataRequestFulfilledBytes32(requestId, response, payment, success);
+        return payment;
+    }
+
+    function fulfillDataRequestBytes(
+        uint256 requestId,
+        bytes memory response,
+        RequestCommitment memory rc,
+        bool isDirectPayment
+    ) external nonReentrant returns (uint256) {
+        uint256 startGas = gasleft();
+        validateDataResponse(rc, requestId);
+        RequestResponseConsumerBase rr;
+        bytes memory resp = abi.encodeWithSelector(
+            rr.rawFulfillDataRequestBytes.selector,
+            requestId,
+            response
+        );
+        bool success = fulfill(resp, rc);
+        uint256 payment = pay(rc, isDirectPayment, startGas);
+        emit DataRequestFulfilledBytes(requestId, response, payment, success);
+        return payment;
+    }
+
+    function validateJobId(bytes32 jobId) internal pure returns (bool) {
```

Comment:
How about we precompute these values and store them in mapping? It could be something similar to what I suggested at `RequestResponseConsumerBase`?